File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.js
function solution(my_string) {
  const result = my_string.split(' ')
  let num = 0;
  for (let i = 0; i < result.length; i++) {
    if(i == 0) num = parseInt(result[i]);
    if(result[i] == '+'){
      num += parseInt(result[i+1]);
      i++;
    }
    if(result[i] == '-'){
      num -= parseInt(result[i+1]);
      i++;
    }
  }
  return num
}

File: 프로그래머스/0/120902. 문자열 계산하기/문자열 계산하기.java
class Solution {
        public int solution(String my_string) {
        String[] result = my_string.split(" ");
        int num = 0;
        num = Integer.parseInt(result[0]);

        for (int i = 1; i < result.length; i++) {
            if (result[i].equals("+")) {
                num += Integer.parseInt(result[i + 1]);
                i++;
            } else if (result[i].equals("-")) {
                num -= Integer.parseInt(result[i + 1]);
                i++;
            }
        }    
        return num;
    }
}